### DISCUSSION TITLE
Sense of `react` and `html` properties

### DISCUSSION THREAD

**Discussion by deadcoder0904** [neg4n/next-api-og-image#14] [General]
I don't think there was a need to make `react` property as `html` could've been used instead.

Now if someone uses `react` but want to use font or tailwind, then they probably need a way to pass `link` & `script` right?

What do you think?

**Comment by neg4n** [comment]
Hey @deadcoder0904, whats the issue with passing `link` and `script` tags?

**Comment by deadcoder0904** [comment]
@neg4n if I'm using `react` inside `template`, where would I pass `link` & `script`? Or do you still allow for me to add a boilerplate like `html` wrapper.

I don't think `react` was needed. See my project → https://github.com/deadcoder0904/got-image-buffer-to-disk/

I've made it to show OG Image Template live so you can use HMR & make changes & it will generate og image accordingly. See `og-image-preview.tsx` file.

**Comment by neg4n** [comment]
> @neg4n if I'm using react inside template, where would I pass link & script? Or do you still allow for me to add a boilerplate like html wrapper.

I've made example code for this scenario https://github.com/neg4n/next-api-og-image/blob/main/example/pages/api/advanced-typescript-react.tsx 😎 

> I don't think react was needed. See my project → deadcoder0904/got-image-buffer-to-disk

In my opinion - `react` was certainly needed in order to maintain `next-api-og-image` easiness of use and simplicity.

By the way @deadcoder0904, I will move this thread to Discussions since it is not actually an issue with the library.

**Reply by deadcoder0904** [reply]
@neg4n idk it looks weird. most people like me associate `react` with having to write components & not edit the `html` file so yeah an `html` tag would've been suffice.

bdw can we pass other types of params like `object` to the react component without converting them to `string`?

if that's possible, maybe then we have 1 difference. not having to write these lines of code → https://github.com/deadcoder0904/got-image-buffer-to-disk/blob/main/pages/api/og-image.ts#L11-L16

**Reply by neg4n** [reply]
> with having to write components & not edit the html file so yeah an html tag would've been suffice.

What do you mean in the part about *html file*? You must either provide react functional component to `react` property or function returning HTML as string to `html` property. Whats weird about this? 

The library is stupid and claims there aren't any *html files* or *components*, splitting your open-graph image generation logic into multiple files is not mandatory or I'd rather say it is optional and fully up to user.

**Reply by deadcoder0904** [reply]
> What do you mean in the part about _html file_? You must either provide react functional component to `react` property or function returning HTML as string to `html` property. Whats weird about this?

i mean they both look eerily similar so i thought the need isn't there to have 2 properties since we have to write the html skeleton anyway if i use react.

what about passing `object`, i dont think it's possible right now? when you add `req.body` support, will it be possible?

**Reply by neg4n** [reply]
> what about passing object, i dont think it's possible right now? when you add req.body support, will it be possible?

Explain please 😇 What object passed where?

**Reply by deadcoder0904** [reply]
like passing `someComplexData` here → https://github.com/neg4n/next-api-og-image/discussions/10#discussioncomment-1476393 instead of passing individual property like `someComplexDataYeah`, `someComplexDataThats`, `someComplexDataFooBar`. makes sense?

**Reply by neg4n** [reply]
@deadcoder0904 https://github.com/neg4n/next-api-og-image/releases/tag/2.1.0
https://github.com/neg4n/next-api-og-image/blob/main/example/pages/api/basic-typescript-react-json.tsx

**Reply by deadcoder0904** [reply]
@neg4n absolutely brilliant. will dive deep this week & try to make it work :)
